src/include/concurrency/timestamp_ordering_transaction_manager.h

```diff
@@ -206,7 +214,10 @@ class TimestampOrderingTransactionManager : public TransactionManager {
    * @param[in]  location     The location
    */
   virtual void PerformUpdate(TransactionContext *const current_txn,
-                             const ItemPointer &location);
+                             const ItemPointer &location,
+                             char *values_buf = nullptr,
+                             uint32_t values_size = 0,
+                             TargetList *offsets = nullptr);
```

Comment:
One does an in-place update while other one creates a new version. It is based on whether the given transaction already has a write lock on it. I think the reason is that If we modify a tuple twice in the same transaction, we can just overwrite the tuple in-place instead of creating a new version.